packages/lodestar/src/network/subnets/attnetsService.ts

```diff
@@ -271,14 +274,18 @@ export class AttnetsService implements IAttnetsService {
 
   /** Trigger a gossip un-subscrition only if no-one is still subscribed */
   private unsubscribeSubnets(subnets: number[], slot: Slot): void {
+    if (this.opts?.subscribeAllSubnets) return;
+
     const forks = getActiveForks(this.config, this.chain.clock.currentEpoch);
     for (const subnet of subnets) {
       if (
         !this.subscriptionsCommittee.isActiveAtSlot(subnet, slot) &&
         !this.subscriptionsRandom.isActiveAtSlot(subnet, slot)
       ) {
         for (const fork of forks) {
-          this.gossip.unsubscribeTopic({type: gossipType, fork, subnet});
+          if (!this.opts?.subscribeAllSubnets) {
+            this.gossip.unsubscribeTopic({type: gossipType, fork, subnet});
+          }
```

Comment:
I suppose we wanna change `shouldProcess` to return true if we have this flag

Reply:
Lighthouse has another flag for it, `--import-all-subnets` since subscribing is a different action that importing the attestation.

However, I noticed that I removed the shouldProcess() call when refactoring the gossip handlers. So I will fix both in another PR

Reply:
Is this check needed if L277 is there?

Reply:
Removed line 277, I think it's more obvious this way